src/host/os_listWindowsRegistry.c

```diff
@@ -0,0 +1,252 @@
+
+	maxNameLength++;
+	maxValueLength++;
+	name = (char*)malloc((size_t)maxNameLength);
+	value = (char*)malloc((size_t)maxValueLength + 1);
+
+	// Iterate over subkeys
+	ok = 1;
+	node.name = name;
+	for (index = 0; index < numSubkeys; index++) {
+		length = maxNameLength;
+		if (RegEnumKeyExA(key, index, name, &length, NULL, NULL, NULL, NULL) != ERROR_SUCCESS) {
+			ok = 0;
+			break;
+		}
+		callback(&node, user);
+	}
+
+	// Iterate over values
+	if (ok) {
+		node.value = value;
+		for (index = 0; index < numValues; index++) {
+			length = maxNameLength;
+			node.valueSize = maxValueLength;
+			if (RegEnumValueA(key, index, name, &length, NULL, &node.type, (LPBYTE)value, &node.valueSize) != ERROR_SUCCESS) {
+				ok = 0;
+				break;
+			}
+
+			// Ensure proper termination of strings (two terminators for the REG_MULTI_SZ)
+			value[node.valueSize] = '\0';
+			value[node.valueSize + 1] = '\0';
+			callback(&node, user);
+		}
+	}
+
+	// Free buffers
+	free(name);
+	free(value);
+
+	return ok;
+}
+
+static void list_callback(const RegNodeInfo* info, void* user)
+{
+	lua_State* L = (lua_State*)user;
+	const char* typeString;
+	
+	// Insert key into the result table (keys are represented as empty tables)
+	if (info->value == NULL) {		
+		lua_createtable(L, 0, 0);
+		lua_setfield(L, -2, info->name);
+		return;
+	}
+
+	// Values are represented as tables containing "type" and "value" records
+	typeString = get_type_string(info->type);
+	lua_createtable(L, 0, 2);
+	lua_pushstring(L, typeString ? typeString : "Unknown");
+	lua_setfield(L, -2, "type");
+
+	switch (info->type)
+	{
+		// Binary encoded values -> size defined string
+		case REG_NONE:
+		case REG_BINARY:
+		case REG_RESOURCE_LIST:
+		case REG_FULL_RESOURCE_DESCRIPTOR:
+		case REG_RESOURCE_REQUIREMENTS_LIST: {
+			lua_pushlstring(L, info->value, info->valueSize);
+			break;
+		}
+
+		// String encoded values -> zero terminated string
+		case REG_SZ:
+		case REG_EXPAND_SZ:
+		case REG_LINK: {
+			lua_pushstring(L, info->value);
+			break;
+		}
+
+		// Numbers
+		case REG_DWORD: {
+			lua_pushinteger(L, *(DWORD32*)info->value);
+			break;
+		}
+
+		case REG_DWORD_BIG_ENDIAN: {
+			lua_pushinteger(L, (info->value[3] << 0) | (info->value[2] << 8) | (info->value[1] << 16) | (info->value[0] << 24));
+			break;
+		}
+
+		case REG_QWORD: {
+			lua_pushinteger(L, *(DWORD64*)info->value);
+			break;
+		}
+
+		// Multiple strings
+		case REG_MULTI_SZ: {
+			DWORD i, j, k;
+
+			lua_newtable(L);
+			for (i = j = 0, k = 1; i < info->valueSize; i++)
+			{
+				if (info->value[i] != 0)
+					continue;
+
+				if (i == j)
+					break;
+
+				lua_pushlstring(L, &info->value[j], i - j);
+				lua_rawseti(L, -2, k);
+				j = i + 1;
+				k++;
+			}
+			break;
+		}
+
+		// Unknown field -> nil
+		default: {
+			lua_pushnil(L);
+			break;
+		}
+	}
+	lua_setfield(L, -2, "value");
+
+	// Complete the value subtable
+	lua_setfield(L, -2, info->name);
+}
+
+int os_listWindowsRegistry(lua_State* L)
```

Comment:
Could this be `os.getWindowsRegistry` instead of `os.listWindowsRegistry`?

All nitpicks I know…

Reply:
I am afraid it could not, since it is allowed to have subkey and value of the same name inside a single key. In this particular case, it is not possible to choose the correct operation just from the path. 

What about calling it 'os.getWindowsRegistryKey'? Or having a second optional boolean argument to select between key and value...